### ISSUE TITLE
poor quality of one random image

### ISSUE THREAD

**Issue by maobenz** [oneThousand1000/EG3D-projector#10]
I just try a random image and i find the quality of inversion is very poor.

Do you have any ideas about it?

<img width="1336" alt="image" src="https://user-images.githubusercontent.com/37300524/196168600-d277060f-09a9-4f62-a110-0acc1ed7d394.png">

**Comment by oneThousand1000** [comment]
Did you align the input image according to https://github.com/NVlabs/eg3d/blob/main/dataset_preprocessing/ffhq/preprocess_in_the_wild.py ?

**Comment by luchaoqi** [comment]
> Did you align the input image according to https://github.com/NVlabs/eg3d/blob/main/dataset_preprocessing/ffhq/preprocess_in_the_wild.py ?

I am also having poor results when following the steps in the original eg3d repo (wild img following preprocessing steps with ffhq pretrained model). The arguments used are all defaults/same as the ones in README. Do you have any ideas to improve the results? Thanks!

test img:

![img00000366](https://user-images.githubusercontent.com/46330265/213952915-8d32ae82-7499-48ec-8cf1-b2d456aebee6.png)

results:

https://user-images.githubusercontent.com/46330265/213952911-8ca9a2a8-b15a-4a5b-9d72-6b12861fdda2.mp4

**Comment by oneThousand1000** [comment]
> > Did you align the input image according to https://github.com/NVlabs/eg3d/blob/main/dataset_preprocessing/ffhq/preprocess_in_the_wild.py ?
> 
> I am also having poor results when following the steps in the original eg3d repo (wild img following preprocessing steps with ffhq pretrained model). The arguments used are all defaults/same as the ones in README. Do you have any ideas to improve the results? Thanks!
> 
> test img:
> 
> ![img00000366](https://user-images.githubusercontent.com/46330265/213952915-8d32ae82-7499-48ec-8cf1-b2d456aebee6.png)
> 
> results:
> 
>  img00000366_w_plus_pretrained.mp4

Hey, I think it is caused by the EG3D model itself and my simple inversion project.
For the EG3D model, the performance on extreme posture is much worse than frontal, it is due to the imbalanced pose distribution in FFHQ. It is still a challenging problem.
For this simple inversion project, the input image only provides information from a single view, and it is hard for PTI to generate full-view results. So I recommend you to use a better inversion method, e.g., https://github.com/jiaxinxie97/HFGI3D

**Comment by oneThousand1000** [comment]
Actually, this repo is just a simple implementation of the projector mentioned in EG3D, not the best choice for projecting an image into EG3D's latent space :)

**Comment by luchaoqi** [comment]
Thanks!

**Comment by oneThousand1000** [comment]
> Thanks! But from the official website, seems they are able to get pretty good results with single img + PTI tho.
> 
>  inversion_compressed.mp4

The input image you use has ear occluded, the input images in the video contain more complete information. This repo cannot generate the regions that are occluded.

You can see the results I generated using my repo: https://github.com/NVlabs/eg3d/issues/28#issuecomment-1159512947, 
here is the input re-aligned image and the input camera parameters:
[01457.zip](https://github.com/NVlabs/eg3d/files/8947809/01457.zip)
![01457](https://user-images.githubusercontent.com/32099648/213961749-8b1af583-8c6e-4dee-82c1-a518a648ebb0.png)

**Comment by luchaoqi** [comment]
Weird that I am getting a slightly different camera matrix than yours after following pytorch_3d_recon:

mine:
```
            [
                0.9982488751411438,
                0.01629943959414959,
                -0.056863944977521896,
                0.14564249100599475,
                0.010219544172286987,
                -0.9943544864654541,
                -0.1056165024638176,
                0.2914214260210597,
                -0.05826440826058388,
                0.10485044121742249,
                -0.9927797317504883,
                2.6802727132270365,
                0.0,
                0.0,
                0.0,
                1.0,
                4.2647,
                0.0,
                0.5,
                0.0,
                4.2647,
                0.5,
                0.0,
                0.0,
                1.0
            ]
```
yours:
```
array([ 0.99852723,  0.01640092, -0.05171374,  0.13343237,  0.01112113,
       -0.9948467 , -0.10077892,  0.27816952, -0.05310011,  0.10005538,
       -0.99356395,  2.6823157 ,  0.        ,  0.        ,  0.        ,
        1.        ,  4.2647    ,  0.        ,  0.5       ,  0.        ,
        4.2647    ,  0.5       ,  0.        ,  0.        ,  1.        ])
```

**Comment by oneThousand1000** [comment]
> Weird that I am getting a slightly different camera matrix than yours after following pytorch_3d_recon:
> 
> mine:
> 
> ```
>             [
>                 0.9982488751411438,
>                 0.01629943959414959,
>                 -0.056863944977521896,
>                 0.14564249100599475,
>                 0.010219544172286987,
>                 -0.9943544864654541,
>                 -0.1056165024638176,
>                 0.2914214260210597,
>                 -0.05826440826058388,
>                 0.10485044121742249,
>                 -0.9927797317504883,
>                 2.6802727132270365,
>                 0.0,
>                 0.0,
>                 0.0,
>                 1.0,
>                 4.2647,
>                 0.0,
>                 0.5,
>                 0.0,
>                 4.2647,
>                 0.5,
>                 0.0,
>                 0.0,
>                 1.0
>             ]
> ```
> 
> yours:
> 
> ```
> array([ 0.99852723,  0.01640092, -0.05171374,  0.13343237,  0.01112113,
>        -0.9948467 , -0.10077892,  0.27816952, -0.05310011,  0.10005538,
>        -0.99356395,  2.6823157 ,  0.        ,  0.        ,  0.        ,
>         1.        ,  4.2647    ,  0.        ,  0.5       ,  0.        ,
>         4.2647    ,  0.5       ,  0.        ,  0.        ,  1.        ])
> ```

Yes, the matrix I uploaded is directly obtained from the dataset.json (in https://github.com/NVlabs/eg3d/blob/main/dataset_preprocessing/ffhq/runme.py), and the image is from the ffhq dataset.
It is ok to use a slightly different matrix.

**Comment by luchaoqi** [comment]
Hello, I have a follow-up question regarding your implementation [here](https://github.com/oneThousand1000/EG3D-projector/blob/5f0340caf47c9c8d1079eac02f7044fb52ca5abb/eg3d/gen_videos_from_given_latent_code.py#L97). You might have also noticed the problem [here](https://github.com/NVlabs/eg3d/issues/28) that feeding optimized latent code directly into the generator without mapping network i.e. no camera information included. 
 
I went through the issue post in the original eg3d repo but didn't find any useful information. Any takeaway conclusion without including camera information in the `ws` from your experiments so far?

**Comment by pfeducode** [comment]
> > 谢谢！但是从官方网站来看，他们似乎能够通过单个 img + PTI 获得相当不错的结果。
> > inversion_compressed.mp4
> 
> 您使用的输入图像有耳朵遮挡，视频中的输入图像包含更完整的信息。此 repo 无法生成被遮挡的区域。
> 
> 你可以看到我使用我的 repo 生成的结果：[NVlabs/eg3d#28 (comment)](https://github.com/NVlabs/eg3d/issues/28#issuecomment-1159512947)， 这里是输入重新对齐的图像和输入相机参数： [01457.zip](https://github.com/NVlabs/eg3d/files/8947809/01457.zip) ![01457](https://user-images.githubusercontent.com/32099648/213961749-8b1af583-8c6e-4dee-82c1-a518a648ebb0.png)

Why does the regenerated image look different from the original image
regenerated image
![400](https://user-images.githubusercontent.com/100210045/230106979-dc121f85-c025-4a87-bc48-2447ea82bb11.png)
source image
![00001](https://user-images.githubusercontent.com/100210045/230107324-d0f3696c-7057-4f1e-b8aa-29d859ec408f.png)

**Comment by pfeducode** [comment]
> 
Thank you for your suggestion
